File: week2/design-hashset.java
class MyHashSet {
    int[] set;
    
    /** Initialize your data structure here. */
    public MyHashSet() {
        set = new int[1000000];
    }
    
    public void add(int key) {
        if (set[key] == 0) {
            set[key]++;
        }
    }
    
    public void remove(int key) {
        if (set[key] > 0) { 
            set[key]--;
        }
    }
    
    /** Returns true if this set contains the specified element */
    public boolean contains(int key) {
        return set[key] > 0;
    }
}

File: summer2021/unit2/design-hashset.py
class MyHashSet:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.array = [0 for i in range(1000000 + 1)]
        

    def add(self, key: int) -> None:
        self.array[key] = 1

    def remove(self, key: int) -> None:
        self.array[key] = 0

    def contains(self, key: int) -> bool:
        """
        Returns true if this set contains the specified element
        """
        return self.array[key] == 1
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)